src/puppetlabs/puppetdb/export.clj

```diff
@@ -0,0 +1,106 @@
+(ns puppetlabs.puppetdb.export
+  (:require [clj-time.core :refer [now]]
+            [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [puppetlabs.kitchensink.core :as kitchensink]
+            [puppetlabs.puppetdb.archive :as archive]
+            [puppetlabs.puppetdb.cheshire :as json]
+            [puppetlabs.puppetdb.catalogs :as catalogs]
+            [puppetlabs.puppetdb.factsets :as factsets]
+            [puppetlabs.puppetdb.reports :as reports]
+            [puppetlabs.puppetdb.utils :as utils]
+            [com.rpl.specter :as sp]
+            [clj-time.format :as time-fmt]
+            [clj-time.coerce :as time-coerce]
+            [puppetlabs.puppetdb.schema :as pls]
+            [schema.core :as s]))
+
+(def export-metadata-file-name "export-metadata.json")
+(def query-api-version :v4)
+(def ^:private command-versions
+  ;; This is not ideal that we are hard-coding the command version here, but
+  ;;  in our current architecture I don't believe there is any way to introspect
+  ;;  on which version of the `replace catalog` matches up with the current
+  ;;  version of the `catalog` endpoint... or even to query what the latest
+  ;;  version of a command is.  We should improve that.
+  {:replace_catalog 6
+   :store_report 5
+   :replace_facts 4})
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Utility Functions
+
+(defn query-child-href-internally
+  [query-fn child-href]
+  (let [[parent-str identifier child-str] (take-last 3 (str/split child-href #"/"))
```

Comment:
I'm gonna leave this for now since it was already this way before, I'll make a note of it in another related ticket.